SwiftUIDZ/SwiftUIDZ/SignUpView.swift

```diff
@@ -0,0 +1,202 @@
+// SignUpView.swift
+// Copyright © RoadMap. All rights reserved.
+
+import SwiftUI
+
+/// Экран входа
+struct SignUpView: View {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let width: CGFloat = 360
+        static let yellowColorName = "pastelYellow"
+        static let orangeColorName = "pastelOrange"
+        static let purpleColorName = "pastelPurple"
+        static let darkPurpleColorName = "pastelDarkPurple"
+        static let redColorName = "pastelRed"
+        static let darkRedColorName = "pastelDarkRed"
+        static let fontSize: CGFloat = 30
+        static let navBarHeight: CGFloat = 10
+        static let fiftyPadding: CGFloat = 50
+        static let zeroSpacing: CGFloat = 0
+        static let cornerRadius: CGFloat = 150
+        static let capsuleHeight: CGFloat = 80
+        static let strokeLineWidth: CGFloat = 5
+        static let logInText = "Log in"
+        static let signUpText = "Sign up"
+        static let twoFloat: CGFloat = 2
+        static let fourFloat: CGFloat = 4
+        static let phoneNumberText = "Phone number"
+        static let fortyPadding: CGFloat = 40
+        static let placeholderPhoneNumberText = "+7 (000) 000-00-00"
+        static let textFieldInlineHeight: CGFloat = 2
+        static let twentyPadding: CGFloat = 20
+        static let passwordText = "Password"
+        static let placeholderPasswordText = "Enter your password"
+        static let maxCountChar = 15
+        static let minCountChar = 6
+        static let fontSizeBig: CGFloat = 34
+        static let seventyPadding: CGFloat = 70
+        static let signUpButtonTitle = "SIGN UP"
+        static let forgotYourPasswordButtonTitle = "Forgot your password?"
+        static let alertText = "Call tech support\n 8 800 555 55 35"
+        static let checkVerificationButtonTitle = "Check verification"
+    }
+
+    // MARK: - Private enum
+
+    private enum FocusedField {
+        case username, password
+    }
+
+    // MARK: - Body
+
+    var body: some View {
+        VStack {
+            GradientNavigationView(navBarHeight: Constants.navBarHeight)
+            capsuleView
+            phoneNumberTextFieldView
+            passwordTextFieldView
+            signUpButtonView
+            forgotYourPasswordButtonView
+            checkVerificationButtonView
+            Spacer()
+        }
+        .onSubmit {
+            focusedField = focusedField == .username ? .password : nil
+        }
+        .onTapGesture {
+            hideKeyboard()
+        }
+        .navigationBarTitleDisplayMode(.inline)
+    }
+
+    // MARK: - Private Properties
+
+    @StateObject private var viewModel = SignUpViewModel()
+    @FocusState private var focusedField: FocusedField?
+    @State private var userPhoneNumber = ""
+    @State private var userPassword = ""
+    @State private var text = ""
+
+    private var checkVerificationButtonView: some View {
```

Comment:
```suggestion
    private var checkVerificationNavLinkView: some View {
```
может это навигейшн линк?